pkg/ipmi/blobs/blob_transfer.go

```diff
@@ -386,14 +393,19 @@ func (h *BlobHandler) BlobStat(id string) (*BlobStats, error) {
 		return nil, err
 	}
 
-	buf := bytes.NewReader(data)
-	var stats BlobStats
+	buf := bytes.NewBuffer(data)
+	var statsHdr blobStatsHdr
 
-	if err := binary.Read(buf, binary.LittleEndian, &stats); err != nil {
+	if err := binary.Read(buf, binary.LittleEndian, &statsHdr); err != nil {
 		return nil, fmt.Errorf("failed to read response: %v", err)
 	}
 
-	return &stats, nil
+	return &BlobStats{
+		State:       statsHdr.State,
+		Size:        statsHdr.Size,
+		MetadataLen: statsHdr.MetadataLen,
+		Metadata:    buf.Bytes(),
```

Comment:
this smells funny to me. this is the entire buffer, header included, isn't it?

Reply:
https://pkg.go.dev/bytes#Buffer.Bytes returns a slice holding the unread portion of the buffer. binary.Read already consumed the header from it.